### PR TITLE
Document optional character parameter for trim functions

### PR DESCRIPTION
**Pull request by mavam**
## Summary
- Updated documentation for `trim()`, `trim_start()`, and `trim_end()` functions to document the optional second parameter
- Added comprehensive integration tests for all three functions
- The functionality has existed since August 2024 but was not documented

## Test plan
- [x] Added integration tests for all three trim functions
- [x] Tests cover whitespace trimming, custom character trimming, and edge cases
- [x] All tests pass with the existing implementation

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: docs/functions/trim.md [hunk 1]
```diff
@@ -4,15 +4,20 @@ category: String/Transformation
 example: '" hello ".trim()'
 ---
 
-Trims whitespace from both ends of a string.
+Trims whitespace or specified characters from both ends of a string.
 
 ```tql
 trim(x:string) -> string
+trim(x:string, chars:string) -> string
```
Comment (IyeOnline) [inline docs/functions/trim.md:11]:
```suggestion
trim(x:string, [chars:string]) -> string
```

File: changelog/changes/field-trait-noted.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+```
+
+This also works with `trim_start()` and `trim_end()` for one-sided trimming:
+
+```tql
+from {
+  start: "///api/v1/users".trim_start("/"),
+  end: "data.csv.tmp.bak".trim_end(".tmpbak")
+}
```
Comment (jachris) [inline changelog/changes/field-trait-noted.md:34]: Doesn't the `chars` argument mean that every single one of those chars is trimmed individually? Not the entire sequence, right?
